### PR TITLE
Use switch statement for repo lookup

### PR DESCRIPTION
**Pull request by ericbottard**
Follow up to #243

### CODE REVIEW

File: gradle/releaser.gradle [hunk 1]
```diff
@@ -124,19 +124,19 @@ if (rootProject.hasProperty("artifactory_publish_password")) {
...
+		case (~/^\d+\.\d+\.\d+$/):
+		case (~/^\d+\.\d+\.\d+\.RELEASE$/):
+			return "libs-release-local";
+		case (~/^\d+\.\d+\.\d+-SNAPSHOT$/):
+		case (~/^\d+\.\d+\.\d+\.BUILD-SNAPSHOT$/):
+			return "libs-snapshot-local";
+		case (~/^\d+\.\d+\.\d+-M\d+$/):
+		case (~/^\d+\.\d+\.\d+-RC\d+$/):
+		case (~/^\d+\.\d+\.\d+\.M\d+$/):
+		case (~/^\d+\.\d+\.\d+\.RC\d+$/):
```
Comment (bsideup) [inline gradle/releaser.gradle:137]:
```suggestion
		case (~/^\d+\.\d+\.\d+(-|\.)M\d+$/):
		case (~/^\d+\.\d+\.\d+(-|\.)RC\d+$/):
```

Comment (bsideup) [reply gradle/releaser.gradle:137]: it took me some time to figure out the difference (`-` vs `.`), hence the suggestion to use `(X|Y)` :)

Comment (ericbottard) [reply gradle/releaser.gradle:137]: I hesitated to add back comments which explain that this supports old vs new scheme. In that particular case, that may be the best way to make the difference stand out, wdyt?

Comment (bsideup) [reply gradle/releaser.gradle:137]: I got an idea: add an example next to every pattern:

```groovy
		case (~/^\d+\.\d+\.\d+-M\d+$/):   // 1.2.3-M1
		case (~/^\d+\.\d+\.\d+-RC\d+$/):  // 1.2.3-RC1
		case (~/^\d+\.\d+\.\d+\.M\d+$/):  // 1.2.3.M1
		case (~/^\d+\.\d+\.\d+\.RC\d+$/): // 1.2.3.RC1
```

**Review by bsideup — APPROVED** [review]

File: gradle/releaser.gradle [hunk 2]
```diff
@@ -124,19 +124,19 @@ if (rootProject.hasProperty("artifactory_publish_password")) {
...
-	]
-	return repos.find {e -> p.version =~ e.key}.value
+	switch (p.version) {
+		case (~/^\d+\.\d+\.\d+$/): // eg 1.2.3, new scheme
+		case (~/^\d+\.\d+\.\d+\.RELEASE$/): // eg 1.2.3.RELEASE, old scheme
+			return "libs-release-local";
+		case (~/^\d+\.\d+\.\d+-SNAPSHOT$/): // eg 1.2.3-SNAPSHOT, new scheme
+		case (~/^\d+\.\d+\.\d+\.BUILD-SNAPSHOT$/): // eg 1.2.3.BUILD-SNAPSHOT, old scheme
+			return "libs-snapshot-local";
+		case (~/^\d+\.\d+\.\d+-M\d+$/): // eg 1.2.3-M4, new scheme
```
Comment (bsideup) [inline gradle/releaser.gradle:134]: nit: could use some alignment, to make it easier to compare char by char

**Comment by simonbasle** [comment]
This has been superseded by the latest changes to the release process, via github actions.

**Pull request closed without merging**